mem0/vector_stores/valkey.py

```diff
@@ -0,0 +1,889 @@
+                    "hash": payload.get("hash", f"hash_{id}"),  # Use a default hash if not provided
+                    "memory": payload.get("data", f"data_{id}"),  # Use a default data if not provided
+                    "created_at": int(datetime.fromisoformat(payload["created_at"]).timestamp()),
+                    "embedding": np.array(vector, dtype=np.float32).tobytes(),
+                }
+
+                # Add optional fields
+                for field in ["agent_id", "run_id", "user_id"]:
+                    if field in payload:
+                        hash_data[field] = payload[field]
+
+                # Add metadata
+                hash_data["metadata"] = json.dumps({k: v for k, v in payload.items() if k not in excluded_keys})
+
+                # Store in Valkey
+                self.client.hset(key, mapping=hash_data)
+                logger.debug(f"Successfully inserted vector with ID {id}")
+            except KeyError as e:
+                logger.error(f"Error inserting vector with ID {id}: Missing required field {e}")
+            except Exception as e:
+                logger.exception(f"Error inserting vector with ID {id}: {e}")
+                raise
+
+    def _build_search_query(self, knn_part, filters=None):
+        """
+        Build a search query string with filters.
+
+        Args:
+            knn_part (str): The KNN part of the query.
+            filters (dict, optional): Filters to apply to the search.
+
+        Returns:
+            str: The complete search query string.
+        """
+        # No filters, just use the KNN search
+        if not filters or not any(value is not None for key, value in filters.items()):
+            return f"*=>{knn_part}"
+
+        # Build filter expression
+        filter_parts = []
+        for key, value in filters.items():
+            if value is not None:
+                # Use the correct filter syntax for Valkey
+                filter_parts.append(f"@{key}:{{{value}}}")
+
+        # No valid filter parts
+        if not filter_parts:
+            return f"*=>{knn_part}"
+
+        # Combine filter parts with proper syntax
+        filter_expr = " ".join(filter_parts)
+        return f"{filter_expr} =>{knn_part}"
+
+    def _execute_search_with_fallback(self, query, params):
+        """
+        Execute a search query with fallback options if the initial query fails.
+
+        Args:
+            query (str): The search query to execute.
+            params (dict): The query parameters.
+
+        Returns:
+            The search results.
+        """
+        try:
+            # Execute the search
+            return self.client.ft(self.collection_name).search(query, query_params=params)
+        except ResponseError as e:
+            logger.warning(f"Search error: {e}")
+
+            # Try alternative syntax if the first attempt fails
+            if "Invalid filter expression" in str(e):
+                logger.warning("Trying alternative filter syntax")
```

Comment:
Why are we trying alternative query syntaxes? Valkey search only has one query syntax.